### PR TITLE
Fix lecture edit page if active term is not set

### LINKED ISSUE #682: Lecture edit page not working when no active term is not set
If `Term.active` is false, then calling the `lectures#edit` action results in an exception.

### PR DESCRIPTION
**Pull request by fosterfarrell9**
This fixes #682 by adding the missing case in the `older_than?` method in the `Lecture` model.

### CODE REVIEW

**Review by Splines — CHANGES_REQUESTED** [review]
LGTM.

- Maybe you can add an end-to-end test to ensure that the lecture#edit page looks as expected? (instead of showing an error?)
- And maybe even add a unit test for the `lecture.stale?` method to ensure it gives proper results even in the case where we just have a lecture and not an active term set.

**Comment by fosterfarrell9** [comment]
I added unit tests for `lecture#stale?`, made `lecture#older_than?` private (it is not used anywhere except by the lecture#stale? method) and added a cypress test for the lecture edit page that should fail when the page throws an error.

File: spec/cypress/e2e/lecture_spec.cy.js
Comment (Splines) [inline spec/cypress/e2e/lecture_spec.cy.js:10]:
```suggestion
    cy.createUserAndLogin("teacher").then((teacher) => {
      FactoryBot.create("lecture", "with_teacher_by_id",
        { teacher_id: teacher.id }).as("lecture");
    });
```

File: spec/cypress/e2e/lecture_spec.cy.js
Comment (Splines) [inline spec/cypress/e2e/lecture_spec.cy.js:14]: LGTM, this will definitely fail if the page throws an error ;)

File: app/models/lecture.rb
Comment (Splines) [inline app/models/lecture.rb:833]: Maybe one last thing: could you add a docstring including an explanation for what "stale" means? (doesn't have to include every detail, just a high-level overview maybe)

File: spec/models/lecture_spec.rb
Comment (Splines) [inline spec/models/lecture_spec.rb:100]:
```suggestion
```

**Review by Splines — CHANGES_REQUESTED** [review]

File: spec/models/lecture_spec.rb
Comment (Splines) [inline spec/models/lecture_spec.rb:90]:
```suggestion
```

**Review by Splines — CHANGES_REQUESTED** [review]

File: spec/cypress/e2e/lecture_spec.cy.js
Comment (Splines) [inline spec/cypress/e2e/lecture_spec.cy.js:1]: rename to `lectures_spec.cy.js` (plural) to be consistent with other cypress spec files

**Review by Splines — CHANGES_REQUESTED** [review]

File: spec/cypress/e2e/lecture_spec.cy.js
Comment (fosterfarrell9) [reply spec/cypress/e2e/lecture_spec.cy.js:1]: Thanks for catching this!

**Review by Splines — APPROVED** [review]
LGTM. Note that you'll have to approve the tests such that they can run through before merging via `Squash and merge`.

**Comment by fosterfarrell9** [comment]
run tests

**Pull request merged**
